packages/flutter/test/cupertino/tab_scaffold_test.dart

```diff
@@ -395,6 +395,72 @@ void main() {
     expect(tester.getRect(find.byType(Placeholder)), Rect.fromLTWH(0, 0, 800, 400));
     expect(MediaQuery.of(innerContext).padding.bottom, 0);
   });
+
+  testWidgets('Deleting tabs after selecting them works', (WidgetTester tester) async {
+    final List<int> tabsBuilt = <int>[];
+
+    BottomNavigationBarItem tabGenerator(int index) {
+      return BottomNavigationBarItem(
+        icon: const ImageIcon(TestImageProvider(24, 24)),
+        title: Text('Tab ${index + 1}'),
+      );
+    }
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: CupertinoTabScaffold(
+          tabBar: CupertinoTabBar(
+            items: List<BottomNavigationBarItem>.generate(4, tabGenerator),
+            onTap: (int newTab) => selectedTabs.add(newTab),
+          ),
+          tabBuilder: (BuildContext context, int index) {
+            tabsBuilt.add(index);
+            return Text('Page ${index + 1}');
+          },
+        ),
+      ),
+    );
+
+    expect(tabsBuilt, <int>[0]);
+    // We never manually tapped anything.
+    expect(selectedTabs, <int>[]);
+
+    await tester.tap(find.text('Tab 4'));
+    await tester.pump();
+
+    // Both tabs are built but only one is onstage.
+    expect(tabsBuilt, <int>[0, 0, 3]);
```

Comment:
These numbers seem confusing....what is the significance of 0, 0, 3?

Reply:
The list keeps accumulating from previous interactions. I just cleared the list after asserting now.